### PR TITLE
Enable even more rules

### PR DESCRIPTION
**Pull request by Evangelink**
cc @MarcoRossignoli

### CODE REVIEW

File: src/Microsoft.TestPlatform.AdapterUtilities/TestIdProvider.cs [hunk 1]
```diff
@@ -14,9 +14,9 @@ public class TestIdProvider
     internal const int DigestBytes = DigestBits / 8;
 
     private Guid _id = Guid.Empty;
-    private byte[] _hash = null;
```
Comment (Evangelink) [inline src/Microsoft.TestPlatform.AdapterUtilities/TestIdProvider.cs:17]: @MarcoRossignoli There you go!

**Review by MarcoRossignoli — APPROVED** [review]

**Comment by MarcoRossignoli** [comment]
we're missing only the null ref annotations...maybe we can start for new files/types and put some effort project by project...it could be a long task.

**Comment by Evangelink** [comment]
I have a branch for it but that's pretty big effort. One solution could be to go with enable nullable and annotate all existing file with the disable so that every new added file follows nullable pattern and we could work on enabling it partially on existing files at a slower pace.

**Comment by Evangelink** [comment]
@nohwnd Is it ok to squash and merge here?

**Comment by MarcoRossignoli** [comment]
> I have a branch for it but that's pretty big effort. One solution could be to go with enable nullable and annotate all existing file with the disable so that every new added file follows nullable pattern and we could work on enabling it partially on existing files at a slower pace.

I think it's good...it's a great improvement anyway.

**Pull request merged**